pkg/projectfile/projectfile.go

```diff
@@ -939,6 +963,11 @@ func createCustom(params *CreateParams, lang language.Language) (*Project, error
 			q.Set("branch", params.BranchName)
 		}
 
+		// Remove this block in DX-2307.
+		if params.LegacyCommitID != "" {
```

Comment:
No. In particular when running `state init`, this will be empty. After the project is created on the Platform, the commitID is polled and filled in retroactively.